azure-iot-device/azure/iot/device/common/pipeline/pipeline_stages_mqtt.py

```diff
@@ -240,18 +243,15 @@ def _on_mqtt_disconnected(self, cause=None):
             op = self._pending_connection_op
             self._pending_connection_op = None
 
+            # Swallow any errors, because we intended to disconnect - even if something went wrong, we
+            # got to the state we wanted to be in!
             if cause:
-                # Only create a ConnnectionDroppedError if there is a cause,
-                # i.e. unexpected disconnect.
-                try:
-                    six.raise_from(errors.ConnectionDroppedError, cause)
-                except errors.ConnectionDroppedError as e:
-                    op.error = e
+                logger.warning(
```

Comment:
>logger [](start = 16, length = 6)

log the contents of the error also please.

Reply:
Going to switch this to use logger.exception

---
In reply to: [326735425](https://github.com/Azure/azure-iot-sdk-python/pull/309#discussion_r326735425) [](ancestors = 326735425)